throttle/throttledRecorder.go

```diff
@@ -95,7 +105,10 @@ func (throttler *ThrottledRecorder) StartRecording() error {
 		return throttler.recorder.StartRecording()
 	} else {
 		throttler.recording = false
-		log.Print("Recording throttled")
+		log.Print("Recording not started - currently throttled")
+		if throttler.listener != nil {
```

Comment:
Go allows you to call methods on typed nils so you can avoid these checks by handling the nil case in WhenThrottled. 
This avoids the need to check if throttler.listener is nil everywhere, making the call sites cleaner.

Example: https://play.golang.org/p/X2JoOiN9YUN